src/Form/MediaLibraryFileUploadForm.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace Drupal\stanford_media\Form;
+
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\ReplaceCommand;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\ElementInfoManagerInterface;
+use Drupal\Core\Render\RendererInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\Url;
+use Drupal\dropzonejs\DropzoneJsUploadSaveInterface;
+use Drupal\media\MediaInterface;
+use Drupal\media_library\Form\FileUploadForm;
+use Drupal\media_library\MediaLibraryUiBuilder;
+use Drupal\media_library\OpenerResolverInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Overrides media library file upload form to extend its functionality.
+ *
+ * @package Drupal\stanford_media\Form
+ */
+class MediaLibraryFileUploadForm extends FileUploadForm {
+
+  /**
+   * DropzoneJs upload save service.
+   *
+   * @var \Drupal\dropzonejs\DropzoneJsUploadSaveInterface
+   */
+  protected $dropzoneSave;
+
+  /**
+   * Current active session.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * {@inheritDoc}}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('media_library.ui_builder'),
+      $container->get('element_info'),
+      $container->get('renderer'),
+      $container->get('file_system'),
+      $container->get('dropzonejs.upload_save'),
+      $container->get('current_user'),
+      $container->get('media_library.opener_resolver')
+    );
+  }
+
+  /**
+   * {@inheritDoc}}
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, MediaLibraryUiBuilder $library_ui_builder, ElementInfoManagerInterface $element_info, RendererInterface $renderer, FileSystemInterface $file_system, DropzoneJsUploadSaveInterface $dropzone_upload, AccountProxyInterface $current_user, OpenerResolverInterface $opener_resolver = NULL) {
+    parent::__construct($entity_type_manager, $library_ui_builder, $element_info, $renderer, $file_system, $opener_resolver);
+    $this->dropzoneSave = $dropzone_upload;
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * {@inheritDoc}}
+   */
+  protected function buildInputElement(array $form, FormStateInterface $form_state) {
+    $element = parent::buildInputElement($form, $form_state);
+
+    $element['container']['dropzone'] = [
```

Comment:
Dropzone is not handling multiple file uploads. Only one file is added.
https://share.getcloudapp.com/Z4uzLrnm

![](https://p21.f4.n0.cdn.getcloudapp.com/items/Z4uzLrnm/Screen+Recording+2019-10-14+at+02.40+PM.gif?v=31525c35f5e62eaa76cf1e2be51da3eb)

Reply:
in WYSIWYG?

Reply:
WYSWYG and Card field so far from what I have tested.

Reply:
Also, it would be nice if this form was available at http://cardinalsites.loc/media/add/image, http://cardinalsites.loc/media/file as well.

Reply:
WSYIWYG and card fields have a cardinality of 1. if you add a field that allows unlimited images/files, then you can upload more

Reply:
`/admin/content/media/add/bulk` provides the dropzonejs form

Reply:
http://cardinalsites.loc/admin/content/media/add/bulk works.

Reply:
The language in the drop field is plural and might have to be tweaked.